src/wallet-user/protection/AccountProtectionFrame.tsx

```diff
@@ -0,0 +1,254 @@
+import { Col, Row } from 'react-bootstrap';
+
+import Frame from "src/common/Frame";
+import { GREEN, ORANGE, YELLOW, rgbaToHex } from "src/common/ColorTheme";
+import Icon from "src/common/Icon";
+import { useLogionChain } from "src/logion-chain";
+import ButtonGroup from "src/common/ButtonGroup";
+import Button from "src/common/Button";
+import { useCallback } from "react";
+import { useUserContext } from "../UserContext";
+import ExtrinsicSubmissionStateView from "src/ExtrinsicSubmissionStateView";
+import RecoveryRefusal from "./RecoveryRefusal";
+import SelectLegalOfficer from "./SelectLegalOfficer";
+import { useCommonContext } from "src/common/CommonContext";
+
+export type ProtectionRecoveryRequestStatus = 'pending' | 'accepted' | 'activated' | 'unavailable' | 'rejected';
+
+export type Refusal = 'single' | 'double' | 'none';
+
+export interface Props {
+    type: ProtectionRecoveryRequestStatus;
+}
+
+export default function AccountProtectionFrame(props: Props) {
+    const { submitCall, clearSubmissionState, extrinsicSubmissionState } = useLogionChain();
+    const { availableLegalOfficers } = useCommonContext();
+    const { protectionState, activateProtection, claimRecovery } = useUserContext();
+
+    const activateProtectionCallback = useCallback(async () => {
+        try {
+            await submitCall(activateProtection!);
+        } catch(e) {
+            console.log(e);
+        } finally {
+            clearSubmissionState();
+        }
+    }, [ submitCall, activateProtection, clearSubmissionState ]);
+
+    const claimRecoveryCallback = useCallback(async () => {
+        try {
+            await submitCall(claimRecovery!);
+        } catch(e) {
+            console.log(e);
+        } finally {
+            clearSubmissionState();
+        }
+    }, [ submitCall, claimRecovery, clearSubmissionState ]);
+
+    if(protectionState === undefined || availableLegalOfficers === undefined) {
+        return null;
+    }
+
+    const protectionParameters = protectionState.protectionParameters;
+    const legalOfficer1 = protectionParameters.legalOfficers[0];
+    const legalOfficer2 = protectionParameters.legalOfficers[1];
+    let legalOfficer1Status: ProtectionRequestStatus;
+    let legalOfficer2Status: ProtectionRequestStatus;
+    if(protectionState instanceof PendingRecovery || protectionState instanceof RejectedRecovery) {
+        legalOfficer1Status = protectionParameters.states[0].status;
+        legalOfficer2Status = protectionParameters.states[1].status;
+    } else if(props.type === 'accepted') {
+        legalOfficer1Status = 'ACCEPTED';
+        legalOfficer2Status = 'ACCEPTED';
+    } else {
+        legalOfficer1Status = 'ACTIVATED';
+        legalOfficer2Status = 'ACTIVATED';
+    }
+
+    let refusal: Refusal = 'none';
+    if (props.type === 'rejected') {
+            refusal = protectionState.protectionParameters.states[1].status === 'REJECTED' ? 'double' : 'single';
+    }
```

Comment:
Note for later: `refusal` should be an attribute of `RejectedRecovery`.

Reply:
Good point, I'll create a technical debt for that.